### ISSUE TITLE
Try overload resolution ambiquity

### ISSUE THREAD

**Issue by EliasJorgensen** [vavr-io/vavr-kotlin#17]
When i use `Try.getOrElseThrow`, i get the following error:
![image](https://user-images.githubusercontent.com/3519438/76628073-74c91600-653c-11ea-8d8a-8fea570aa88d.png)

This is the code in question:
```kotlin
val flow = Try { flowDao.save(Flow(source.name, source.applicationId)) }
	.getOrElseThrow { FlowNameAlreadyInUse() }
```

This is my exception:
```kotlin
@ResponseStatus(HttpStatus.CONFLICT)
class FlowNameAlreadyInUse : RuntimeException()
```

I am using it the same way that i would throw an exception with `java.util.Optional.getOrElseThrow`, so i believe this to be a bug.

I can make it work by putting the exception in a lambda, but i believe this to be a bad solution, especially given that IntelliJ keeps suggesting me to remove the useless lambda.
```kotlin
val flow = Try { flowDao.save(Flow(source.name, source.applicationId)) }
	.getOrElseThrow { -> FlowNameAlreadyInUse() }
```

**Comment by mincong-h** [comment]
Hi @EliasJorgensen , I never used Kotlin before and just tried it. It's cool 😎 . I can reproduce the problem (#18). I think this issue should go to vavr core (https://github.com/vavr-io/vavr) because vavr-kotlin is only a wrapper. It seems that Kotlin cannot distinguish function and supplier when the methods are overloaded. For Java Optional, it works because it has only two overloaded methods: `orElseThrow()` and `orElseThrow(Supplier)`, which are not ambiguous. cc @danieldietrich, @ruslansennov 

TL;DR: I think there is no easy and short term solution here

**Comment by EliasJorgensen** [comment]
Hi @mincong-h, thanks for checking out my bug report. I'm a bit unsure whether you want me to report the bug in the core repository, as there is no easy "fix"? Let me know and i'll create one 😄

**Comment by mincong-h** [comment]
@EliasJorgensen , sorry I was not clear. I think there is nothing to do on your side. Let's wait and see if Daniel, Ruslan, or other members have some thoughts about this.
